### PR TITLE
Fetch CR asynchronously in AbstractOperator reconcile processing

### PR DESCRIPTION
**Pull request by MikeEdgar**
### Type of change

_Select the type of your PR_

- Bugfix
- ~~Enhancement / new feature~~
- Refactoring
- ~~Documentation~~

### Description

- Fetch CR asynchronously to avoid blocking event loop thread, adjust test mocks
- Refactor create/update and delete cases to separate methods.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Comment by MikeEdgar** [comment]
The diff is a bit friendlier with white-space changes ignored.

**Comment by scholzj** [comment]
@MikeEdgar Is this ready for review?

**Comment by MikeEdgar** [comment]
> @MikeEdgar Is this ready for review?

@scholzj yes, please

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaBridgeAssemblyOperatorTest.java [hunk 1]
```diff
@@ -605,6 +605,8 @@ public void testReconcileCallsCreateOrUpdate(VertxTestContext context) {
         // when requested ConfigMap for a specific Kafka Bridge cluster
         when(mockBridgeOps.get(eq(kbNamespace), eq("foo"))).thenReturn(foo);
         when(mockBridgeOps.get(eq(kbNamespace), eq("bar"))).thenReturn(bar);
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaBridgeAssemblyOperatorTest.java:607]: Both here and in the other classes -> do we still need to mock the `get()` methods? Or would `getAsync()` be enough?

Comment (MikeEdgar) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaBridgeAssemblyOperatorTest.java:607]: `getAsync` may be enough. I'll try it and update the PR if they're not necessary.

Comment (MikeEdgar) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaBridgeAssemblyOperatorTest.java]: Sync `get` methods have been removed.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaStatusTest.java [hunk 2]
```diff
@@ -185,16 +184,18 @@ public void testPauseReconciliationsStatus(VertxTestContext context) throws Pars
...
-
-            assertThat(kafkaCaptor.getValue(), is(notNullValue()));
-            assertThat(kafkaCaptor.getValue().getStatus(), is(notNullValue()));
-            KafkaStatus status = kafkaCaptor.getValue().getStatus();
-
-            assertThat(status.getConditions().size(), is(1));
-            assertThat(status.getConditions().get(0).getStatus(), is("True"));
-            assertThat(status.getConditions().get(0).getType(), is("ReconciliationPaused"));
-            assertThat(status.getObservedGeneration(), is(1L));
+            context.verify(() -> {
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaStatusTest.java:187]: As you changed this already ... you should be able to use something like this and replace the result success check as well:

```
.onComplete(context.succeeding(pfa -> context.verify(() -> { ... })))
```

**Review by scholzj — COMMENTED** [review]
I left one nit. But it LGTM otherwise. I will also run the regresison tests.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaStatusTest.java (hunk 2 above)
Comment (MikeEdgar) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaStatusTest.java]: Updated to use `context.succeeding`.

**Review by scholzj — APPROVED** [review]
Thanks for the PR.

**Comment by MikeEdgar** [comment]
Rebased and merge conflicts resolved.

**Review by ppatierno — APPROVED** [review]
LGTM

**Review by tombentley — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the PR @MikeEdgar

**Pull request merged**